README.adoc

```diff
@@ -198,6 +198,68 @@ include::{common-includes}/devmode-build.adoc[]
 Check out the service that you created at the
 http://localhost:9080/LibertyProject/System/properties[^] URL. 
 
+// =================================================================================================
+// Exception Handling
+// =================================================================================================
+
+== Exception Handling
+
+If a resource doesn't exist or the service experiences an exception, the client will receive either an empty response or an HTML error page respectively.
+A client should not receive either of these.
+Clients should receive JSON responses. 
+
+This will be handled using an `ExceptionMapper` which maps exceptions to responses and builds appropriate JSON payloads. 
+
+[role="code_command hotspot file=0", subs="quotes"]
+----
+#Create the `ErrorResponse` class.#
+`src/main/java/io/openliberty/guides/rest/exceptions/ErrorResponse.java`
+----
+
+ErrorResponse.java
+[source, Java, linenums, role="code_column hide_tags=comment"]
+----
+include::finish/src/main/java/io/openliberty/guides/rest/exceptions/ErrorResponse.java[]
+----
+
+The [hotspot file=0]`ErrorResponse` class will define the JSON contents of our response. 
+It contains fields for the error code and error message. 
+
+[role="code_command hotspot file=1", subs="quotes"]
+----
+#Create the `GenericExceptionMapper` class.#
+`src/main/java/io/openliberty/guides/rest/exceptions/GenericExceptionMapper.java`
+----
+
+GenericExceptionMapper.java
+[source, Java, linenums, role="code_column hide_tags=comment"]
+----
+include::finish/src/main/java/io/openliberty/guides/rest/exceptions/GenericExceptionMapper.java[]
+----
+
+When JAX-RS sees any `Throwable`, it will search for a matching `ExceptionMapper` interface that has been marked with the [hotspot=provider file=1]`@Provider` annotation. 
+It will pass the `Throwable` to the [hotspot=toResponse file=1]`toResponse()` method to construct a response using the [hotspot file=0]`ErrorResponse` class.
+
+The [hotspot file=1]`GenericExceptionMapper` class will be used for any `Throwable` that does not have a more specific `ExceptionMapper` interface implemented. 
+
+[role="code_command hotspot file=2", subs="quotes"]
+----
+#Create the `NotFoundExceptionMapper` class.#
+`src/main/java/io/openliberty/guides/rest/exceptions/NotFoundExceptionMapper.java`
+----
+
+NotFoundExceptionMapper.java
```

Comment:
We discussed in slack that it might be better to use a different exception type here - ideally an application-specific, business exception (like `WidgetNotFoundException`) since `NotFoundException` is a subclass of `WebApplicationException` - and it could hold it's own `Response` object (and thus doesn't need to be mapped).